apps/omg_bus/lib/omg_bus/pubsub.ex

```diff
@@ -36,16 +36,30 @@ defmodule OMG.Bus.PubSub do
 
   defmacro __using__(_) do
     quote do
+      alias OMG.Bus.Event
```

Comment:
hmmm, the problem with this approach (in general in this module) is that you're coupling the logic (business logic - the information about rootchain, childchain etc.) with something that was entirely "clean" from that perspective.

Reply:
I wonder if there's a better approach

Reply:
I have the same feeling about coupling those. Macros are magic to me and I don't know if it's possible to do, but maybe we could introduce a macro in a module outside of `:omg_bus`, possibly in `:omg`, that expands a struct so it matches previous API. Then this macro would be used in modules where we call `subscribe` or `broadcast`.

Reply:
perhaps there's a simpler approach. maybe the information to a topic (like childchain) is where that data is being generated. so the processes that are generating childchain data can publish events, but it goes through the interface of a childchain module that appends this data, similarly for watcher or rootchain.

Reply:
I'll try with that tomorrow.

Reply:
I decided to follow @pnowosie advice as it decouples  business names from `:omg_bus` quite elegantly. I tried other approaches, but this seems best.